forte/processors/data_augment/algorithms/skeleton_op.py

```diff
@@ -0,0 +1,765 @@
+            self._replaced_annos[pid].add((Span(pos, pos), inserted_text))
+            self._inserted_annos_pos_len[pid][pos] = len(inserted_text)
+            return True
+        return False
+
+    def delete_annotations(self, input_anno: Annotation) -> bool:
+        r"""
+        This is a utility function to delete an annotation. If the same
+        annotation is tried to be deleted twice, the function will terminate
+        and return False.
+
+        Args:
+            input_anno: The annotation to remove.
+        Returns:
+            A bool value. True if the deletion happened, False otherwise.
+        """
+        pid: int = input_anno.pack.pack_id
+        if self._overlap_with_existing(pid, input_anno.begin, input_anno.end):
+            return False
+
+        self._replaced_annos[pid].add((input_anno.span, ""))
+        self._deleted_annos_id[pid].add(input_anno.tid)
+        return True
+
+    def replace_annotations(
+        self, replacement_anno: Annotation, replaced_text
+    ) -> bool:
+        r"""
+        This is a utility function to record specifically a replacement of an annotation.
+        With this function, an annotation can be replaced with another annotation.
+        If the same annotation is tried to be replaced twice, the function will terminate
+        and return False.
+
+        Args:
+            input_anno: The annotation to replace.
+        Returns:
+            A bool value. True if the replacement happened, False otherwise.
+        """
+        pid: int = replacement_anno.pack.pack_id
+        if self._overlap_with_existing(
+            pid, replacement_anno.begin, replacement_anno.end
+        ):
+            return False
+
+        self._replaced_annos[pid].add(
+            (Span(replacement_anno.begin, replacement_anno.end), replaced_text)
+        )
+        return True
+
+    @abstractmethod
+    def augment(self, data_pack: DataPack) -> bool:
+        r"""
+        This method is left to be implemented by the user of this Op.
+        The user can use any of the given utility functions to perform
+        augmentation.
+
+        Args:
+            `input_anno`: the input annotation to be replaced.
+
+        Returns:
+            A boolean value indicating if the augmentation
+            was successful (True) or unsuccessful (False).
+        """
+        raise NotImplementedError
+
+    @classmethod
+    def default_configs(cls):
+        """
+        Returns:
+            A dictionary with the default config for this processor.
+        Following are the keys for this dictionary:
+            - `augment_entry`:
+                Defines the entry the processor will augment.
+                It should be a full qualified name of the entry class.
+                For example, "ft.onto.base_ontology.Sentence".
+        """
+        return {
+            "augment_entry": "ft.onto.base_ontology.Token",
+            "other_entry_policy": {},
```

Comment:
We don't need a default value for `augment_entry` right? We could in fact create a default value for `other_entry_policy`. Here we also miss the documentation for `other_entry_policy`

Reply:
I think `augment_entry` is an important default config since it determines on what type of ontology will the augmentation happen. Sorry about the `other_entry_policy documentation. I'll add it.

Reply:
And tbh, we don't know whether this op is only concerning one type of `augment_entry`, so having this `augment_entry` key here is probably meaningless, I think we can remove this key entirely.